ddl/ddl_api.go

```diff
@@ -6627,7 +6628,12 @@ func (d *ddl) AlterTableCache(ctx sessionctx.Context, ti ast.Ident) (err error)
 
 	err = d.doDDLJob(ctx, job)
 	err = d.callHookOnChanged(err)
-	return errors.Trace(err)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	_, err = ctx.(sqlexec.SQLExecutor).ExecuteInternal(context.Background(), "insert ignore into mysql.table_cache_meta values (%?, 'NONE', 0, 0)", t.Meta().ID)
```

Comment:
What if `ExecuteInternal` fails for some network reason but the ddl succeed?

Reply:
OK, I'll move this before the execution of acural DDL, and if this result in error, the DDL fails...